pkg/event/event_test.go

```diff
@@ -307,23 +313,28 @@ func TestEventHandler_ProcessEvents(t *testing.T) {
 			case c := <-logOut:
 				s1 := strings.Index(c, "[")
 				s2 := strings.Index(c, "]")
-				rs := strings.Replace(c, c[s1+1:s2], "0", -1)
 
-				if strings.HasPrefix(c, string(test.processName)) {
-					assert.Equal(t, test.wantProcessState, rs, test.desc)
-				}
-				if strings.HasPrefix(c, "GM[") {
-					assert.Equal(t, test.wantGMState, rs, test.desc)
+				if s1 != -1 && s2 > s1 {
+					c = c[:s1+1] + "0" + c[s2:]
 				}
-				if strings.HasPrefix(c, "ptp4l[") {
-					assert.Equal(t, test.wantClockState, rs, test.desc)
+				// Mark expected log as seen
+				for want := range expectedLogs {
+					if want == "" || c == want {
+						expectedLogs[want] = true
+					}
 				}
 			default:
+				time.Sleep(100 * time.Millisecond)
```

Comment:
I don't want to derail the PR, but I'd love to learn more about why we're sleeping here.

Every time I see a sleep I see a red flag; trauma from past jobs I've held.

Reply:
Sleeping allows other go-routines to run which is especially important for the tests at they may run in environments for where the thread/process count is low :)